modules/StaticISO15118VASProvider/iso15118_vas/ISO15118_vasImpl.cpp

```diff
@@ -0,0 +1,149 @@
+    auto param_name = json["name"].get<std::string>();
+    auto param_type = json["type"].get<std::string>();
+
+    types::iso15118_vas::Parameter parameter;
+    parameter.name = param_name;
+
+    EVLOG_debug << "    Parameter name: " << param_name;
+
+    if (param_type == "bool") {
+        parameter.value.bool_value = json["value"].get<bool>();
+        EVLOG_debug << "      Parameter value: " << (parameter.value.bool_value.value() ? "true" : "false");
+    } else if (param_type == "int32") {
+        parameter.value.int_value = json["value"].get<int32_t>();
+        EVLOG_debug << "      Parameter value: " << std::to_string(parameter.value.int_value.value());
+    } else if (param_type == "int16") {
+        parameter.value.short_value = json["value"].get<int16_t>();
+        EVLOG_debug << "      Parameter value: " << std::to_string(parameter.value.short_value.value());
+    } else if (param_type == "int8") {
+        parameter.value.byte_value = json["value"].get<int8_t>();
+        EVLOG_debug << "      Parameter value: " << std::to_string(parameter.value.byte_value.value());
+    } else if (param_type == "string") {
+        parameter.value.finite_string = json["value"].get<std::string>();
+        EVLOG_debug << "      Parameter value: " << parameter.value.finite_string.value();
+    } else if (param_type == "rational") {
+        parameter.value.rational_number = json["value"].get<float>();
+        EVLOG_debug << "      Parameter value: " << std::to_string(parameter.value.rational_number.value());
+    } else {
+        EVLOG_AND_THROW(Everest::EverestConfigError("Unknown parameter type: " + param_type));
+    }
+
+    return parameter;
+}
+
+/**
+ * @brief Convert a single JSON ParameterSet Object (id and parameters) to a ParameterSet object. Also checks for
+ * duplicate parameter names.
+ * @param json The JSON object to convert.
+ * @return The converted ParameterSet object.
+ */
+types::iso15118_vas::ParameterSet parameter_set_from_json(const nlohmann::ordered_json& json) {
+    auto set_id = json["id"].get<uint16_t>();
+    auto parameters_raw = json["parameters"];
+
+    std::vector<types::iso15118_vas::Parameter> parameters;
+    std::set<std::string> parameter_names; // only used for duplicate check
+
+    EVLOG_debug << "  Parameter set ID: " << set_id;
+    for (const auto& parameter : parameters_raw) {
+        auto param = parameter_from_json(parameter);
+        parameters.push_back(param);
+
+        // check for duplicate parameter names
+        if (parameter_names.find(param.name) != parameter_names.end()) {
+            EVLOG_AND_THROW(Everest::EverestConfigError(
+                "Parameter name " + param.name + " already exists in parameter set with id " + std::to_string(set_id)));
+        }
+
+        parameter_names.insert(param.name);
+    }
+
+    return types::iso15118_vas::ParameterSet{
+        .set_id = set_id,
+        .parameters = parameters,
```

Comment:
Unfortunately, this may not be used here, as EVerest uses C++17 and this is a C++20 feature. Change this to:
```
types::iso15118_vas::ParameterSet{set_id, parameters}
```